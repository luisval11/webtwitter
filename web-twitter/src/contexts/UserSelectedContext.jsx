import React from "react";

export const UserSelectedContext = React.createContext(null);
