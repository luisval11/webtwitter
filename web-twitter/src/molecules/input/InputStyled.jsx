import React from "react";
import PropTypes from "prop-types";
import { Form } from "react-bootstrap";
import { FormLabelStyled } from "../../atoms/FormLabelStyled";

export const InputStyled = ({
	index,
	className,
	handleInput,
	label,
	name,
	...rest
}) => {
	return (
		<Form.Group controlId={index} className={className}>
			{label && <FormLabelStyled htmlFor={name} text={label} />}
			<Form.Control
				id={name}
				onChange={(event) => {
					handleInput && handleInput(event);
				}}
				{...rest}
			/>
		</Form.Group>
	);
};

InputStyled.propTypes = {
	index: PropTypes.any,
	className: PropTypes.string,
	handleInput: PropTypes.func,
	label: PropTypes.string,
	name: PropTypes.string,
	rest: PropTypes.object,
};

export default InputStyled;
