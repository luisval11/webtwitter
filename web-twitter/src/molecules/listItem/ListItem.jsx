import React, { useContext } from "react";
import PropTypes from "prop-types";
import { ListGroupItem } from "react-bootstrap";
import { UserSelectedContext } from "../../contexts/UserSelectedContext";
import "./style.css";
import { TypedListItem } from "./TypedListItem";

export const ListItem = ({ item, type, onClickAction, onClickItem }) => {
	const { selectedUser } = useContext(UserSelectedContext);

	return (
		<ListGroupItem
			role="listitem"
			className={`list-item${
				item.id === selectedUser && type === "user"
					? " list-item--selected"
					: ""
			}`}
			onClick={() =>
				onClickItem !== undefined &&
				onClickItem !== null &&
				onClickItem(item.id)
			}
		>
			<TypedListItem item={item} onClickAction={onClickAction} type={type} />
		</ListGroupItem>
	);
};

ListItem.propTypes = {
	item: PropTypes.object.isRequired,
	type: PropTypes.string.isRequired,
	onClickAction: PropTypes.func,
	onClickItem: PropTypes.func,
};

export default ListItem;
