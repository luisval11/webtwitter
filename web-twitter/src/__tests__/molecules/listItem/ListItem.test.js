import { mount } from "enzyme";
import { UserSelectedContext } from "../../../contexts/UserSelectedContext";
import { InputStyled } from "../../../molecules/input/InputStyled";
import ListItem from "../../../molecules/listItem/ListItem";

describe("Test on ListItem", () => {
	const itemUser = {
		id: 1,
		name: "Alice",
		isFollowed: true,
	};
	const itemPost = {
		id: 1,
		nameAuthor: "Alice",
		authorId: 1,
		content: "Lorem ipsum content large",
		createdDate: new Date(2022, 2, 28, 14, 30),
	};
	const onClickAction = () => {
		console.log("Dummy click action");
	};
	const onClickItem = () => {
		console.log("Dummy click item");
	};

	test("should match initial snapshot for TypedListItem 'user'", () => {
		let wrapper = mount(
			<UserSelectedContext.Provider value={{ selectedUser: null }}>
				<ListItem
					type={"user"}
					item={itemUser}
					onClickAction={onClickAction}
					onClickItem={onClickItem}
				/>
			</UserSelectedContext.Provider>
		);
		expect(wrapper).toMatchSnapshot();
	});

	test("should match an snapshot for TypedListItem 'user' with that user being selected", () => {
		const { id } = itemUser;
		let wrapper = mount(
			<UserSelectedContext.Provider value={{ selectedUser: id }}>
				<ListItem
					type={"user"}
					item={itemUser}
					onClickAction={onClickAction}
					onClickItem={onClickItem}
				/>
			</UserSelectedContext.Provider>
		);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("div.list-item--selected").length).toBe(1);
	});

	test("should have 1 'span' containing the value of the attribute 'name'", () => {
		const { name } = itemUser;
		let wrapper = mount(
			<UserSelectedContext.Provider value={{ selectedUser: null }}>
				<ListItem
					type={"user"}
					item={itemUser}
					onClickAction={onClickAction}
					onClickItem={onClickItem}
				/>
			</UserSelectedContext.Provider>
		);
		expect(wrapper.find("span").length).toBe(1);
		expect(wrapper.find("span").text().trim()).toBe(name);
	});

	test("should have 1 'button' with text relating to the attribute 'isFollowed'", () => {
		const { isFollowed } = itemUser;
		let wrapper = mount(
			<UserSelectedContext.Provider value={{ selectedUser: null }}>
				<ListItem
					type={"user"}
					item={itemUser}
					onClickAction={onClickAction}
					onClickItem={onClickItem}
				/>
			</UserSelectedContext.Provider>
		);
		expect(wrapper.find("button").length).toBe(1);
		expect(wrapper.find("button").text().trim()).toBe(
			isFollowed ? "Unfollow" : "Follow"
		);
	});

	test("should match initial snapshot for TypedListItem 'post'", () => {
		let wrapper = mount(
			<UserSelectedContext.Provider value={{ selectedUser: null }}>
				<ListItem type={"post"} item={itemPost} onClickItem={onClickItem} />
			</UserSelectedContext.Provider>
		);
		expect(wrapper).toMatchSnapshot();
	});

	test("should have 1 'div' with className 'card-title' containing the value of the attribute 'nameAuthor'", () => {
		const { nameAuthor } = itemPost;
		let wrapper = mount(
			<UserSelectedContext.Provider value={{ selectedUser: null }}>
				<ListItem type={"post"} item={itemPost} onClickItem={onClickItem} />
			</UserSelectedContext.Provider>
		);
		expect(wrapper.find("div.card-title").length).toBe(1);
		expect(wrapper.find("div.card-title").text().trim()).toBe(nameAuthor);
	});

	test("should have 1 'p' with className 'card-text' containing the value of the attribute 'content'", () => {
		const { content } = itemPost;
		let wrapper = mount(
			<UserSelectedContext.Provider value={{ selectedUser: null }}>
				<ListItem type={"post"} item={itemPost} onClickItem={onClickItem} />
			</UserSelectedContext.Provider>
		);
		expect(wrapper.find("p.card-text").length).toBe(1);
		expect(wrapper.find("p.card-text").text().trim()).toBe(content);
	});
});
