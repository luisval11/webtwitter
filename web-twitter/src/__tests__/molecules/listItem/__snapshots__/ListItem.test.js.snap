// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on ListItem should match an snapshot for TypedListItem 'user' with that user being selected 1`] = `
<div
  className="list-item list-item--selected list-group-item"
  onClick={[Function]}
  role="listitem"
>
  <div
    className="user"
  >
    <span>
      Alice
    </span>
    <button
      className="btn btn-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Unfollow
    </button>
  </div>
</div>
`;

exports[`Test on ListItem should match initial snapshot for TypedListItem 'post' 1`] = `
<div
  className="list-item list-group-item"
  onClick={[Function]}
  role="listitem"
>
  <div
    className="post card"
  >
    <div
      className="card-body"
    >
      <div
        className="post--title"
      >
        <div
          className="m-0 card-title h5"
        >
          Alice
        </div>
        <span>
          28 mar
        </span>
      </div>
      <p
        className="mb-0 card-text"
      >
        Lorem ipsum content large
      </p>
    </div>
  </div>
</div>
`;

exports[`Test on ListItem should match initial snapshot for TypedListItem 'user' 1`] = `
<div
  className="list-item list-group-item"
  onClick={[Function]}
  role="listitem"
>
  <div
    className="user"
  >
    <span>
      Alice
    </span>
    <button
      className="btn btn-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Unfollow
    </button>
  </div>
</div>
`;
